addons/pos_gift_card/static/src/js/GiftCardPopup.js

```diff
@@ -37,40 +39,43 @@ odoo.define("pos_gift_card.GiftCardPopup", function (require) {
       this.state.showGiftCardDetails = true;
     }
 
-    addGiftCardProduct(giftCard) {
+    async addGiftCardProduct(giftCard) {
       let gift =
         this.env.pos.db.product_by_id[
           this.env.pos.config.gift_card_product_id[0]
         ];
-      this.env.pos.get_order().add_product(gift, {
-        price: this.state.amountToSet,
-        quantity: 1,
-        merge: false,
-        generated_gift_card_ids: giftCard ? giftCard.id : false,
-      });
+
+      let can_be_sold = !giftCard.buy_pos_order_line_id && !giftCard.buy_line_id;
```

Comment:
don't you get an error when there is no giftcard passed? Because False doesn't have buy_pos_order_line_id attribute

Reply:
No everything is working properly (https://watch.screencastify.com/v/nU0YXZJleD80W0bPjsv6)